### PR TITLE
ci: Add markdownlint GH workflow

### PR DESCRIPTION
**Pull request by spetrosi**
Enhancement: Add markdownlint GH workflow and config file to check all .md files

Reason: Ensure that all markdown files have correct syntax and will be build correctly.

### CODE REVIEW

File: roles/pcp/README.md [hunk 1]
```diff
@@ -88,9 +88,9 @@ Basic PCP setup with monitoring suited for a single host.
   roles:
     - role: performancecopilot.metrics.pcp
       vars:
-        pcp_pmlogger_interval: 10
-        pcp_optional_agents: [dm, nfsclient, openmetrics]
-        pcp_explicit_labels:
+    ### pcp_pmlogger_interval: 10
+    ### pcp_optional_agents: [dm, nfsclient, openmetrics]
+    ### pcp_explicit_labels:
           environment: production
```
Comment (richm) [inline roles/pcp/README.md:94]: not sure what markdownlint was complaining about here, but these should not be commented out

File: roles/pcp/README.md [hunk 2]
```diff
@@ -101,10 +101,10 @@ Central PCP setup for monitoring of several remote hosts.
...
     - role: performancecopilot.metrics.pcp
       vars:
-        pcp_pmlogger_interval: 10
-        pcp_pmlogger_discard: 5
-        pcp_target_hosts: [slip, slop, slap]
-        pcp_rest_api: true
+    ### pcp_pmlogger_interval: 10
+    ### pcp_pmlogger_discard: 5
+    ### pcp_target_hosts: [slip, slop, slap]
+    ### pcp_rest_api: true
```
Comment (richm) [inline roles/pcp/README.md:107]: not sure what markdownlint was complaining about here, but these should not be commented out

File: roles/spark/README.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ Setup export of PCP metrics to a Spark worker thread.
   roles:
     - role: performancecopilot.metrics.spark
       vars:
-        spark_export_port: 44388
+    ### spark_export_port: 44388
```
Comment (richm) [inline roles/spark/README.md:56]: not sure what markdownlint was complaining about here, but these should not be commented out

File: roles/pcp/README.md (hunk 1 above)
Comment (natoscott) [reply roles/pcp/README.md:94]: @spetrosi I think the change is fine for all except those last three lines @richm has called out - this is an example and I think these have accidentally been turned into comments here?

File: roles/pcp/README.md (hunk 2 above)
Comment (natoscott) [reply roles/pcp/README.md:107]: Agreed, this looks like the same issue as above and the change seems incorrect here.

File: roles/spark/README.md (hunk 3 above)
Comment (natoscott) [reply roles/spark/README.md:56]: Yep, same again here - this is an example, and should not be commented like this.

**Review by natoscott — CHANGES_REQUESTED** [review]
Most changes fine, but looks like a handful of accidentally introduced comment lines.

File: roles/pcp/README.md (hunk 1 above)
Comment (spetrosi) [reply roles/pcp/README.md]: You're right, it was some incorrect mass replace on my side. Now it's fixed.

**Comment by spetrosi** [comment]
> Most changes fine, but looks like a handful of accidentally introduced comment lines.

# marks a title in markdown. The more #s, the more title is indented.

**Review by natoscott — APPROVED** [review]
LGTM, thanks!

**Pull request merged**